pkg/agent/clean/adunmigration/rtbs.go

```diff
@@ -158,34 +161,64 @@ func updateCRTB(crtbInterface v3norman.ClusterRoleTemplateBindingInterface, oldC
 		Steps:    10,
 	}
 
+	var createdCrtb *v3.ClusterRoleTemplateBinding
 	err := wait.ExponentialBackoff(backoff, func() (finished bool, err error) {
-		_, err = crtbInterface.Create(newCrtb)
+		// Note: this does NOT trigger an RBAC sync correctly
+		createdCrtb, err = crtbInterface.Create(newCrtb)
 		if err != nil {
 			if apierrors.IsInternalError(err) {
-				logrus.Errorf("[%v] internal error while creating crtb, will backoff and retry: %v", migrateCrtbsOperation, err)
+				logrus.Errorf("[%v] internal error while creating CRTB, will backoff and retry: %v", migrateCrtbsOperation, err)
 				return false, err
 			}
 			return true, fmt.Errorf("[%v] unable to create new CRTB: %w", migrateCrtbsOperation, err)
 		}
 		return true, nil
 	})
 	if err != nil {
-		return fmt.Errorf("[%v] permanent error when creating crtb, giving up: %v", migrateCrtbsOperation, err)
+		return fmt.Errorf("[%v] permanent error when creating CRTB, giving up: %v", migrateCrtbsOperation, err)
 	}
 
 	err = wait.ExponentialBackoff(backoff, func() (finished bool, err error) {
+		// Note: this DOES trigger an RBAC sync which removes the binding in downstream clusters
+		// It does NOT automatically re-queue the newly created binding from above
 		err = crtbInterface.DeleteNamespaced(oldCrtb.Namespace, oldCrtb.Name, &metav1.DeleteOptions{})
 		if err != nil {
 			if apierrors.IsInternalError(err) {
-				logrus.Errorf("[%v] internal error while deleting crtb, will backoff and retry: %v", migrateCrtbsOperation, err)
+				logrus.Errorf("[%v] internal error while deleting CRTB, will backoff and retry: %v", migrateCrtbsOperation, err)
 				return false, err
 			}
 			return true, fmt.Errorf("[%v] unable to delete old CRTB: %w", migrateCrtbsOperation, err)
 		}
 		return true, nil
 	})
 	if err != nil {
-		return fmt.Errorf("[%v] permanent error when deleting crtb, giving up: %v", migrateCrtbsOperation, err)
+		return fmt.Errorf("[%v] permanent error when deleting CRTB, giving up: %v", migrateCrtbsOperation, err)
+	}
+
+	// Adding the annotation here has the side effect of triggering a downstream RBAC sync for our newly created
+	// binding, which is why we perform it last
+
+	newAnnotations[adGUIDMigrationAnnotation] = oldCrtb.UserPrincipalName
+
+	err = wait.ExponentialBackoff(backoff, func() (finished bool, err error) {
+		updatedCrtb, err := crtbInterface.GetNamespaced(oldCrtb.Namespace, createdCrtb.Name, metav1.GetOptions{})
+		if err != nil {
+			return true, fmt.Errorf("[%v] unable to get newly created CRTB: %w", migrateCrtbsOperation, err)
+		}
+
+		updatedCrtb.Annotations = newAnnotations
+		_, err = crtbInterface.Update(updatedCrtb)
```

Comment:
`newAnnotations` is created before reloading the object. That means that something else adding an annotation, like a mutating webhook, will have that annotation removed. Should principal name be merged into the annotations instead of overwriting them? This applies to each of these update situations.